### PR TITLE
feat(as-4190): add out of time shutter page

### PR DESCRIPTION
**Pull request by demzzydev**
## Ticket Number

https://pins-ds.atlassian.net/browse/AS-3736

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [x] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/forms-web-app/src/views/full-planning/out-of-time-shutter-page.njk [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+{% block pageTitle %}
+  {{ title }}
+{% endblock %}
+
+{% block content %}
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+      <h1 class="govuk-heading-l">You cannot appeal.</h1>
+      <p class="govuk-body">Your deadline to appeal has passed</p>
```
Comment (shilpabasu22) [inline packages/forms-web-app/src/views/full-planning/out-of-time-shutter-page.njk:13]: Full stop is missing at the end of line

File: packages/forms-web-app/src/views/full-planning/out-of-time-shutter-page.njk [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      <p class="govuk-body">Your deadline was 10 October 2021. This is 6 months from the decision date.</a>.<p>
```
Comment (shilpabasu22) [inline packages/forms-web-app/src/views/full-planning/out-of-time-shutter-page.njk:14]: There are 2 full stops here and why do we have </a> tag? Also, can there be a way for me to uniquely identify this element? May be a data-cy tag?

Comment (demzzydev) [reply packages/forms-web-app/src/views/full-planning/out-of-time-shutter-page.njk]: resolved.

File: packages/forms-web-app/src/views/full-planning/out-of-time-shutter-page.njk (hunk 1 above)
Comment (demzzydev) [reply packages/forms-web-app/src/views/full-planning/out-of-time-shutter-page.njk]: resolved

File: packages/e2e-tests/cypress/integration/full-planning/before-you-start/shutter-page-deadline-passed.feature [hunk 3]
```diff
@@ -0,0 +1,8 @@
+Feature:Shutter page - Deadline passed
+  As an appellant
+  I am notified that I cannot make an appeal when I enter an out of time date
+  so that I am not able to proceed further with my application
+
+  Scenario: AC01 - Shutter page
+    Given an appellant is on the shutter page for date passed for appeal
+    Then appellant is displayed details for  out of time shutter page
```
Comment (swdpcomputing) [inline packages/e2e-tests/cypress/integration/full-planning/before-you-start/shutter-page-deadline-passed.feature:8]: Double space between for and out.

File: packages/e2e-tests/cypress/integration/full-planning/before-you-start/shutter-page-deadline-passed/shutter-page-deadline-passed-steps.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import { verifyPageTitle } from '../../../../support/common/verify-page-title';
+import { verifyPageHeading } from '../../../../support/common/verify-page-heading';
+const pageHeading = 'You cannot appeal.';
+const pageTitle = 'You cannot appeal - Before you start - Appeal a planning decision - GOV.UK';
+const pageUrl='/before-you-start/you-cannot-appeal';
+Given('an appellant is on the shutter page for date passed for appeal',()=>{
+  goToPage(pageUrl);
+});
+
+Then('appellant is displayed details for  out of time shutter page',()=>{
```
Comment (swdpcomputing) [inline packages/e2e-tests/cypress/integration/full-planning/before-you-start/shutter-page-deadline-passed/shutter-page-deadline-passed-steps.js:12]: Double space between for and out.

File: packages/forms-web-app/__tests__/unit/controllers/full-planning/out-of-time-shutter-page.spec.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+const { mockReq, mockRes } = require('../../mocks');
+const { VIEW } = require('../../../../src/lib/views');
+
+jest.mock('../../../../src/lib/logger');
+
+describe('controllers/full-planning/out-of-time-shutter-page', () => {
+  let req;
+  let res;
+
+  beforeEach(() => {
```
Comment (swdpcomputing) [inline packages/forms-web-app/__tests__/unit/controllers/full-planning/out-of-time-shutter-page.spec.js:11]: Remove lines 11 - 16

File: packages/forms-web-app/__tests__/unit/controllers/full-planning/out-of-time-shutter-page.spec.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    req = mockReq();
+    res = mockRes();
+
+    jest.resetAllMocks();
+  });
+
+  describe('getOutOfTimeShutterPage', () => {
```
Comment (swdpcomputing) [inline packages/forms-web-app/__tests__/unit/controllers/full-planning/out-of-time-shutter-page.spec.js:18]: Remove this describe and move under the first describe.

File: packages/forms-web-app/__tests__/unit/routes/full-planning/index.test.js [hunk 7]
```diff
@@ -20,7 +21,8 @@ describe('routes/full-planning/index', () => {
     expect(use).toHaveBeenCalledWith(localPlanningDepartmentRouter);
     expect(use).toHaveBeenCalledWith(typeOfPlanningRouter);
     expect(use).toHaveBeenCalledWith(useADifferentServiceRouter);
+    expect(use).toHaveBeenCalledWith(outOfTimeRouter);
 
-    expect(use.mock.calls.length).toBe(5);
+    expect(use.mock.calls.length).toBe(6);
```
Comment (swdpcomputing) [inline packages/forms-web-app/__tests__/unit/routes/full-planning/index.test.js:26]: Will probably move to 7 with rebase.

**Review by swdpcomputing — CHANGES_REQUESTED** [review]

File: packages/forms-web-app/__tests__/unit/controllers/full-planning/out-of-time-shutter-page.spec.js (hunk 5 above)
Comment (demzzydev) [reply packages/forms-web-app/__tests__/unit/controllers/full-planning/out-of-time-shutter-page.spec.js:11]: doing this breaks the unit tests

File: packages/forms-web-app/__tests__/unit/controllers/full-planning/out-of-time-shutter-page.spec.js (hunk 6 above)
Comment (demzzydev) [reply packages/forms-web-app/__tests__/unit/controllers/full-planning/out-of-time-shutter-page.spec.js:18]: done

File: packages/forms-web-app/__tests__/unit/routes/full-planning/index.test.js (hunk 7 above)
Comment (demzzydev) [reply packages/forms-web-app/__tests__/unit/routes/full-planning/index.test.js:26]: this also breaks unit tests

File: packages/e2e-tests/cypress/integration/full-planning/before-you-start/shutter-page-deadline-passed/shutter-page-deadline-passed-steps.js (hunk 4 above)
Comment (shilpabasu22) [reply packages/e2e-tests/cypress/integration/full-planning/before-you-start/shutter-page-deadline-passed/shutter-page-deadline-passed-steps.js:12]: updated

File: packages/e2e-tests/cypress/integration/full-planning/before-you-start/shutter-page-deadline-passed.feature (hunk 3 above)
Comment (shilpabasu22) [reply packages/e2e-tests/cypress/integration/full-planning/before-you-start/shutter-page-deadline-passed.feature:8]: updated

**Review by swdpcomputing — APPROVED** [review]

**Pull request merged**